### ISSUE TITLE
Redefine field "Geographic Scope" / info_scope

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#691]
Change Geographic Scope to a multi-select instead of a single select. From there you can pick either a list of continents or countries you operate in.

Additional to do

 * Port all networks with "Global" to the list of continents
 * Ask all networks that have set "Regional" to redefine. After a TBD cut-off date set the value to ""

This is a follow-up to #37 

@peeringdb/pc, please make up your mind

**Comment by ynbrthr** [comment]
+1

**Comment by grizz** [comment]
+1

**Comment by ynbrthr** [comment]
defaulting to “” will always be inaccurate, using the org country should at least be true in some instances

**Comment by arnoldnipper** [comment]
> defaulting to “” will be 100% inaccurate, using the org country should at least be true in some instances

You are opening a can of worms, @ynbrthr :) ... 

 * defaulting to org country is at least as worse as setting it to "" (remember that we only do that if the network doesn't react)
 * we could instead look for the cc the RIR assigned the network to. See discussion in #473. But this also does not necessarily mean that this ASN **operates** in this country. 

Hence setting it to "" in case a network had classified it as "Regional" and also defaulting to "" is best we can do.

**Comment by arnoldnipper** [comment]
As we already have a quorum, I would propose to give users an early heads up. Thoughts?

**Comment by ynbrthr** [comment]
> > defaulting to “” will be 100% inaccurate, using the org country should at least be true in some instances
> 
> 
> 
> You are opening a can of worms, @ynbrthr :) ... 

I’m game :)


>  * defaulting to org country is at least as worse as setting it to "" (remember that we only do that if the network doesn't react)
> 
>  * we could instead look for the cc the RIR assigned the network to. See discussion in #473. But this also does not necessarily mean that this ASN **operates** in this country. 

maybe the cc as registered with the RIR is a better option than the country registered in pdb. In any case, my point is:

No network operates in no region (the proposed “” default if an org doesn’t update their region record)
At least some networks operate in the country they said they’re attached too, either in peeringdb or with their RIR

so the latter option seems to me a better default choice

> 
> Hence setting it to "" in case a network had classified it as "Regional" and also defaulting to "" is best we can do.

this is where i’m not convinced, see above. on the other hand, i don’t have any data either way, so maybe it’ll make no actual difference for those networks not updating that field after the change.

**Comment by arnoldnipper** [comment]
> this is where i’m not convinced, see above. on the other hand, i don’t have any data either way, so maybe it’ll make no actual difference for those networks not updating that field after the change.

@ynbrthr, for me "" is as good as "Not disclosed". You simply can't tell. It's sort of Schrödinger's cat. Whereas, when setting it to any value, you open the box. And you better don't :)

We can't force networks to set a value. Hence we also should not set one in case they don't. For whatever reason. Because they don't want or because they don't care.

**Comment by grizz** [comment]
I like to not open boxes of cats... :)

I'd rather have no information than misinformation, so would agree with "" as the value, I just wanted to make sure to present options.

> As we already have a quorum, I would propose to give users an early heads up. Thoughts?

Yes, please.

**Comment by ynbrthr** [comment]
> I like to not open boxes of cats... :)
> 
> 
> 
> I'd rather have no information than misinformation, so would agree with "" as the value, I just wanted to make sure to present options.
>

yeah, i didn’t think that thru. “” makes more sense, let’s keep that box closed :)

**Comment by arnoldnipper** [comment]
**Summary**

Change Geographic Scope to a multi-select instead of a single select. From there you can pick either a list of continents or countries you operate in.

Additional to do

 * Port all networks with "Global" to the list of continents
 * Ask all networks that have set "Regional" to redefine. After a TBD cut-off date set the value to ""
 * Default value for this field also is ""
 * Make the multi-select **comfortable**

This is a follow-up to #37

@peeringdb/pc , please make up your mind, if you **do not agree**. Otherwise, it will  move on to 3 in a week.

**Comment by stillwaxin** [comment]
Some slightly tangental thoughts.. Why can't we generate this information algorithmically (the value of "Geographic Scope" field)? We know what facilities a network are in based on their selection of Private Peering Facilities. We know where those facilities are (in theory). So we can determine what scope network they run based on this data. Algorithm can be along the lines as this:
Global:
 - Network is present in at least 2 facilities in each of 4 separate continents
Continental:
 - Network is present in at least 2 facilities in each of 2 countries on a single continent
International:
 - Network is present in at least 1 facility in each of 2 countries
National:
 - Network is present in at least 3 facilities located only in a single country
Regional:
 - Network is present in at least 2 facilities that are more than 100km apart (this does not match UN Geoscheme obviously, note below)
Local:
 - Network is present in at least 1 facility

This is likely not a perfect algorithm that would match all networks and likely needs tweaking. I'd also be happy to adapt this algorithm to match the UN Geoscheme definitions as well.

Then for each we can have another field that describes more precisely where the networks are. For example if a network matched Continental based on being in France and Spain then the field(s)would display : Continent: Europe; Country: France, Spain or something along those lines. (UX is not my thing)

**Comment by arnoldnipper** [comment]
I don't see what we gain, @stillwaxin? And one of our topmost paradigms is that only the owner adds data about themselves.

Moreover, we detected that "Regional" or "International" does exactly mean nothing. That's the reason we want to get rid of it.

**Comment by stillwaxin** [comment]
Few things. This automates something that is currently manual. This implicitly or explicitly defines what the various geographic types are (including regional and international). This does not add any new data as it only uses the facilities the network operator themselves selected. 

If definitions cannot be made for the various items under Geographic Scope then I would suggest we get rid of the field entirely and have all search and geo-related functions derived from the linked facility data instead.

**Comment by grizz** [comment]
@stillwaxin It's not a bad idea, but it should be a separate issue / field. This one is specifically for networks to be able to set where they operate at for themselves, which may or may not be tied to where they have facilities.

Please make a new issue for that, and we can discuss there. :)

**Comment by grizz** [comment]
@peeringdb/pc this requires advance user notice

**Comment by job** [comment]
a bit late, but +1 :-

**Comment by grizz** [comment]
Also worth noting, this will require adding a field and deprecating the current one until v3.

**Comment by arnoldnipper** [comment]
> @peeringdb/pc this requires advance user notice

Is that worth an additional label "Advance Notice"?
